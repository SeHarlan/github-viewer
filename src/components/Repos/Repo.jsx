import React from 'react';
import PropTypes from 'prop-types';

const Repo = ({ name }) => (
  <h3>{name}</h3>
);

Repo.propTypes = {
  name: PropTypes.string.isRequired
};
export default Repo;
