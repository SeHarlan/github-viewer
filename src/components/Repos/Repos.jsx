import React from 'react';
import PropTypes from 'prop-types';
import Repo from './Repo.jsx';

const Repos = ({ repos }) => {
  const repoElements = repos.map(repo => (
    <li key={repo.name}>
      <Repo {...repo} />
    </li>
  ));
  return (
    <ul>
      {repoElements}
    </ul>
  );
};


Repos.propTypes = {
  repos: PropTypes.array.isRequired
};

export default Repos;
