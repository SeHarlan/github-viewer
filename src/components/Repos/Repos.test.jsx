import React from 'react';
import { shallow } from 'enzyme';
import Repos from './repos';

describe('Repos component', () => {
  it('renders Repos', () => {
    const wrapper = shallow(<Repos repos={[{ repoName: 'repo test' }]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
