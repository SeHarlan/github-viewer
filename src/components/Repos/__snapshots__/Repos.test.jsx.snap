// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repos component renders Repos 1`] = `
<ul>
  <li
    key="repo test"
  >
    <Repo
      repoName="repo test"
    />
  </li>
</ul>
`;
